driver-core/src/main/com/mongodb/internal/client/model/bulk/ConcreteClientNamespacedWriteModel.java

```diff
@@ -17,13 +17,23 @@
 package com.mongodb.internal.client.model.bulk;
 
 import com.mongodb.MongoNamespace;
-import com.mongodb.client.model.bulk.ClientNamespacedWriteModel;
-import com.mongodb.client.model.bulk.ClientWriteModel;
+import com.mongodb.client.model.bulk.ClientNamespacedDeleteManyModel;
+import com.mongodb.client.model.bulk.ClientNamespacedDeleteOneModel;
+import com.mongodb.client.model.bulk.ClientNamespacedInsertOneModel;
+import com.mongodb.client.model.bulk.ClientNamespacedReplaceOneModel;
+import com.mongodb.client.model.bulk.ClientNamespacedUpdateManyModel;
+import com.mongodb.client.model.bulk.ClientNamespacedUpdateOneModel;
 
 /**
  * This class is not part of the public API and may be removed or changed at any time.
  */
-public final class ConcreteClientNamespacedWriteModel implements ClientNamespacedWriteModel {
+public final class ConcreteClientNamespacedWriteModel
```

Comment:
Does this implementation technique necessitate a Javadoc comment in `ClientNamespacedWriteModel`, some sort of warning that you shouldn't attempt to cast the return value to any other type, even though the value returned may implement other types?

Reply:
And if so, maybe it's worth biting the bullet and just creating the six simple sub-classes of this class, one for each interface.

Reply:
Done in 6f01e61b750dd1dadebe75d8f635cc52992bb7a6.